packages/orchestration/src/examples/restake.kit.js

```diff
@@ -0,0 +1,262 @@
+ */
+
+export const restakeInvitaitonGuardShape = {
+  Restake: M.call(ChainAddressShape, RepeaterOptsShape).returns(M.promise()),
+  CancelRestake: M.call().returns(M.promise()),
+};
+
+/**
+ * @param {Zone} zone
+ * @param {object} opts
+ * @param {VowTools} opts.vowTools
+ * @param {ZCF} opts.zcf
+ * @param {Remote<TimerService>} opts.timer
+ * @param {ReturnType<typeof prepareRestakeWaker>} opts.makeRestakeWaker
+ * @param {RestakeParams} opts.params
+ */
+export const prepareRestakeHolderKit = (
+  zone,
+  {
+    vowTools: { watch, asVow },
+    zcf,
+    timer,
+    makeRestakeWaker,
+    params: { minimumDelay, minimumInterval },
+  },
+) => {
+  return zone.exoClassKit(
+    'RestakeHolderKit',
+    {
+      invitationMakers: M.interface('InvitationMakers', {
+        ...restakeInvitaitonGuardShape,
+      }),
+      holder: M.interface('Holder', {
+        restake: M.call(ChainAddressShape, RepeaterOptsShape).returns(VowShape),
+        cancelRestake: M.call().returns(VowShape),
+      }),
+    },
+    /**
+     * @param {OrchestrationAccount<any> & StakingAccountActions} orchAccount
+     */
+    orchAccount => {
+      return /** @type {RestakeHolderState} */ (
+        harden({
+          orchAccount,
+          restakeRepeater: undefined,
+        })
+      );
+    },
+    {
+      invitationMakers: {
+        /**
+         * @param {CosmosValidatorAddress} validator
+         * @param {RepeaterOpts} opts
+         * @returns {Promise<Invitation>}
+         */
+        Restake(validator, opts) {
+          trace('Restake', validator, opts);
+          return zcf.makeInvitation(
+            seat => {
+              seat.exit();
+              return watch(this.facets.holder.restake(validator, opts));
+            },
+            'Restake',
+            undefined,
+            EmptyProposalShape,
+          );
+        },
+        CancelRestake() {
+          trace('Cancel Restake');
+          return zcf.makeInvitation(
+            seat => {
+              seat.exit();
+              return watch(this.facets.holder.cancelRestake());
+            },
+            'Cancel Restake',
+            undefined,
+            EmptyProposalShape,
+          );
+        },
+      },
+      holder: {
+        /**
+         * @param {CosmosValidatorAddress} validator
+         * @param {RepeaterOpts} opts
+         * @returns {Vow<TimestampRecord>}
+         */
+        restake(validator, opts) {
+          trace('restake', validator, opts);
+          return asVow(() => {
+            mustMatch(
+              validator,
+              ChainAddressShape,
+              'invalid validator address',
+            );
+            mustMatch(opts, RepeaterOptsShape, 'invalid repeater options');
+            const { delay, interval } = opts;
+            delay >= minimumDelay ||
+              Fail`delay must be at least ${minimumDelay}`;
+            interval >= minimumInterval ||
+              Fail`interval must be at least ${minimumInterval}`;
```

Comment:
Is there a reason  not to use TimeMath here?

Reply:
No - thanks for bringing it to attention. The latest plan for https://github.com/Agoric/agoric-sdk/issues/9796 doesn't seem to depend on this PR, but will make sure to use `TimeMath` when this PR is picked up again.